### ISSUE TITLE
Bad `IsDualSemigroupRep` implication

### ISSUE THREAD

**Issue by fingolfin** [semigroups/Semigroups#885]
This code is problematic:
```gap
InstallTrueMethod(IsDualSemigroupRep,
                  IsSemigroup and IsDualSemigroupElementCollection);
```
A representation is type information / encodes the internal layout of an object; it is not a mathematical property and as such should not be implied in this way.

This is in a way a return of the problem from issue #556 and is the main blocker for the GAP PR https://github.com/gap-system/gap/pull/3006

**Comment by james-d-mitchell** [comment]
Thanks for the report @fingolfin but I'm fairly sure we've fixed this already in:

a19419587bcbd0e0cdd5cc22b2cdd394119cd575

Despite the misleading name, `IsDualSemigroupRep` is a category not a representation, and will be renamed in the next major release of Semigroups. 

Maybe I need to make a release to incorporate the change in the commit mentioned above, I can do that!

**Comment by fingolfin** [comment]
Ah, sorry, the above was based on a CI run from a few weeks (?) ago, so perhaps there was already a newer release that contains the fix.

Sorry for not checking more carefully!

**Comment by james-d-mitchell** [comment]
No problem, I'm not sure I've released a version with that change yet, but I can do it this week.

**Comment by fingolfin** [comment]
There is no need to hurry. I plan to release GAP 4.12.1 this week, but I don't think there is a need to include this new release of semigroups; if it is "too late" it'll just be in 4.12.2 which I right now hope to release in mid-December

**Comment by james-d-mitchell** [comment]
Resolved in 5.1.0
